delfin/drivers/dell_emc/vnx/vnx_block/component_handler.py

```diff
@@ -565,3 +572,340 @@ def list_storage_hosts(self, storage_id):
+                archive_file_list, resources_map, start_time, end_time)
+            if not performance_lines_map:
+                LOG.warn("The required performance data was not found!")
+                return metrics
+            metrics = self.create_metrics(storage_id, resource_metrics,
+                                          resources_map, resources_type_map,
+                                          performance_lines_map)
+        except exception.DelfinException as err:
+            err_msg = "Failed to collect metrics from VnxBlockStor: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise err
+        except Exception as err:
+            err_msg = "Failed to collect metrics from VnxBlockStor: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+        finally:
+            self._remove_archive_file(archive_file_list)
+        return metrics
+
+    def create_metrics(self, storage_id, resource_metrics, resources_map,
+                       resources_type_map, performance_lines_map):
+        metrics = []
+        for resource_obj in resources_type_map.keys():
+            if not resources_map.get(resource_obj) \
+                    or not resources_type_map.get(resource_obj):
+                continue
+            resources_type = resources_type_map.get(resource_obj)
+            labels = {
+                'storage_id': storage_id,
+                'resource_type': resources_type,
+                'resource_id': resources_map.get(resource_obj),
+                'type': 'RAW',
+                'unit': ''
+            }
+            if not performance_lines_map.get(resource_obj):
+                continue
+            metric_model_list = self._get_metric_model(
+                resource_metrics.get(resources_type), labels,
+                performance_lines_map.get(resource_obj),
+                consts.RESOURCES_TYPE_TO_METRIC_CAP.get(
+                    resources_type), resources_type)
+            if metric_model_list:
+                metrics.extend(metric_model_list)
+        return metrics
+
+    def _get__archive_file(self, start_time, end_time):
+        archive_file_list = []
+        archives = self.navi_handler.get_archives()
+        tools = Tools()
+        for archive_info in (archives or []):
+            collection_timestamp = tools.time_str_to_timestamp(
+                archive_info.get('collection_time'), consts.TIME_PATTERN)
+            if collection_timestamp > start_time:
+                archive_file_list.append(archive_info.get('archive_name'))
+            if collection_timestamp > end_time:
+                break
+        return archive_file_list
+
+    def _get_metric_model(self, metric_list, labels, metric_values, obj_cap,
+                          resources_type):
+        metric_model_list = []
+        if not metric_values:
```

Comment:
This `if` statement is already judged outside this function, it's redundant here.

Reply:
Modified